tests/lib/rules/prop-types.js

```diff
@@ -2366,6 +2386,38 @@ ruleTester.run('prop-types', rule, {
       }]
     }, {
       code: [
+        'var Hello = function(props) {',
+        '  return <div>{props.firstname}</div>;',
+        '};',
+        'Hello.propTypes = {}'
+      ].join('\n'),
+      options: [{skipUndeclared: true}],
+      parserOptions: parserOptions,
+      errors: [{
+        message: '\'firstname\' is missing in props validation',
+        line: 2,
+        column: 22
+      }]
+    }, {
+      code: [
+        'class Hello extends React.Component {',
+        '  static get propTypes() {',
```

Comment:
I copied how the other static propTypes were declared in this file. upon closer inspection, this file is inconsistent in how it declares static proptypes. I see three different types:

`static get propTypes() {`
`static propTypes = {`
`static propTypes: {`

do you think I should introduce a fourth type? i'm also happy to consolidate them but I think that is outside the scope of this PR